src/codegen/oa_hash_table_proxy.cpp

```diff
@@ -0,0 +1,275 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// oa_hash_table_proxy.cpp
+//
+// Identification: src/codegen/oa_hash_table_proxy.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "codegen/oa_hash_table_proxy.h"
+
+namespace peloton {
+namespace codegen {
+
+//===----------------------------------------------------------------------===//
+// Return the LLVM type that matches the memory layout of our HashTable
+//===----------------------------------------------------------------------===//
+llvm::Type *OAHashTableProxy::GetType(CodeGen &codegen) {
+  static const std::string kHashTableTypeName = "peloton::OAHashTable";
+  // Check if the hash table type has been registered/cached in the module
+  // already
+  auto *hash_table_type = codegen.LookupTypeByName(kHashTableTypeName);
+  if (hash_table_type != nullptr) {
+    return hash_table_type;
+  }
+
+  // Define and register the type
+  std::vector<llvm::Type *> layout = {
+      // HashEntry * (the open-addressing buckets array)
+      OAHashEntryProxy::GetType(codegen)->getPointerTo(),
+
+      // The number of buckets
+      codegen.Int64Type(),
+
+      // The bucket mask (used for modulo calculation)
+      codegen.Int64Type(),
+
+      // The number of occupied buckets
+      codegen.Int64Type(),
+
+      // The number of entries in the hash table
+      codegen.Int64Type(),
+
+      // The resize threshold
+      codegen.Int64Type(),
+
+      // The entry size (in bytes)
+      codegen.Int64Type(),
+
+      // The key size (in bytes)
+      codegen.Int64Type(),
+
+      // The value size (in bytes)
+      codegen.Int64Type()};
+  hash_table_type = llvm::StructType::create(codegen.GetContext(), layout,
+                                             kHashTableTypeName);
+  return hash_table_type;
+}
+
+//===----------------------------------------------------------------------===//
+// INIT PROXY
+//===----------------------------------------------------------------------===//
+
+const std::string &OAHashTableProxy::_Init::GetFunctionName() {
+  static const std::string kInitFnName =
+#ifdef __APPLE__
+      "_ZN7peloton7codegen5utils11OAHashTable4InitEmmm";
```

Comment:
same as the one in buffering_coosumer.cpp, if peloton does not support OSX, then no need for such macro, similar issue in subsequent lines of this file.